Automattic/wp-calypso
# Help Center: Add logic for archived conversations

Fixes: https://github.com/Automattic/dotcom-forge/issues/9574

## Proposed Changes

| Recent | Archived |
| ------------- | -----------|
|<img width="413" alt="CleanShot 2024-10-29 at 17 27 14@2x" src="https://github.com/user-attachments/assets/7181ec98-8251-4024-914b-f5f74791b16a">| <img width="413" alt="CleanShot 2024-10-29 at 17 28 56@2x" src="https://github.com/user-attachments/assets/a72a9952-654d-48ed-84a1-405f511720cf"> |

* Add logic for Help Center archived conversations.
* Refactor recent/archived conversations logic.
* Removes blue focus color on `Recent` and `Archived` tabs.
<img width="423" alt="CleanShot 2024-10-29 at 17 16 41@2x" src="https://github.com/user-attachments/assets/45aea127-afdd-4421-9002-be22d1141e45">

* Add `createdAt` to metadata when a new conversation is created with Smooch.
* Sort `Recent` conversations by unread messages + last message received. Unread conversations will be at the top of the recent conversations list.

## Why are these changes being made?


* Allow users to view their old conversations and create a separation between older conversations and recent conversations.

## Testing Instructions



* Checkout or use live link
* Go to `/home` with the `?flags=help-center-experience` flag enabled.
* Open the Help Center , click on the recent conversations section, and verify the list is populated correctly.
* To test the archived conversations, you can disable the flag, and modify the smooch.CreateConversation `metadata.createdAt` property ( https://github.com/Automattic/wp-calypso/pull/95839/files#diff-ae6e358ec645887d4afd9f77bb59e2d9aafbf1d71881dfbf823736d594d03b1bR36) to a timestamp from a year ago (https://www.timestamp-converter.com/ - use UTC timestamp) or you can create a new chat and hardcode the conversation `metadata.createdAt` property here: https://github.com/Automattic/wp-calypso/pull/95839/files#diff-3c5495b3c5142d5f12c2d9941b42c29042a6a7a79f64d6c812ab1ef600a45d5bR41.
* Test with a timestamp that is less than a year ago.

## Pre-merge Checklist
